### PR TITLE
nushell: 0.94.1 -> 0.95.0

### PR DESCRIPTION
**Pull request by dairyisscary**
## Description of changes
I re-enabled the tests in the nushell package, only skipping the tests that don't work. I also
captured more of the implementation of `rustPlatform.buildRustPackage`'s checkPhase.

https://www.nushell.sh/blog/2024-06-25-nushell_0_95_0.html

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/shells/nushell/plugins/polars.nix [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+{ stdenv
+, lib
+, rustPlatform
+, openssl
+, nushell
+, pkg-config
+, IOKit
+, Foundation
+, nix-update-script
+,
```
Comment (aidalgol) [inline pkgs/shells/nushell/plugins/polars.nix:10]: If you're going to reformat code, please do so in a separate commit from any other changes.

File: pkgs/shells/nushell/default.nix [hunk 2]
```diff
@@ -13,9 +13,7 @@
 , Security
 , nghttp2
 , libgit2
-# string interpolation dependends on a date that is erroring out
-# this will be fixed in releases after 0.90.1
-, doCheck ? false
+, doCheck ? true
```
Comment (aidalgol) [inline pkgs/shells/nushell/default.nix:16]: Seems to me that we should remove this argument entirely now.

File: pkgs/shells/nushell/default.nix [hunk 3]
```diff
@@ -56,8 +54,17 @@ rustPlatform.buildRustPackage {
...
+    (
+      # The skipped tests all fail in the sandbox because in the nushell test playground,
+      # `$HOME` is unset, so nu fallbacks to looking up the passwd dir of the build user
+      # which is /var/empty, while the expectations do respect $HOME.
+      set -x
+      HOME=$(mktemp -d) cargo test -j $NIX_BUILD_CORES --offline -- \
+        --test-threads=$NIX_BUILD_CORES \
+        --skip=repl::test_config_path::test_default_config_path \
+        --skip=repl::test_config_path::test_xdg_config_bad \
+        --skip=repl::test_config_path::test_xdg_config_empty
```
Comment (aidalgol) [inline pkgs/shells/nushell/default.nix:66]: Is it possible to do this by just setting `cargoTestFlags` and exporting `HOME` in the `preCheck` hook?

**Review by aidalgol — CHANGES_REQUESTED** [review]

File: pkgs/shells/nushell/default.nix (hunk 3 above)
Comment (dairyisscary) [reply pkgs/shells/nushell/default.nix:66]: i really wanted to write this:
```nix
{
  preCheck = ''
    export HOME=$(mktemp -d)
  '';
  checkType = "debug"; # i also tried the default release here
  checkFlags = [
    "--skip=repl::test_config_path::test_default_config_path"
    "--skip=repl::test_config_path::test_xdg_config_bad"
    "--skip=repl::test_config_path::test_xdg_config_empty"
  ];
}
```

but for whatever reason that has hundreds of failures. as far as i can tell the only difference is a `--target` flag:

from `nix log` output:
```
++cargo test -j 12 --target aarch64-apple-darwin --offline -- --test-threads=12 --skip=repl::test_config_path::test_default_config_path --skip=repl::test_config_path::test_xdg_config_bad --skip=repl::test_config_path::test_xdg_config_empty
```
